.github/workflows/validation.yml

```diff
@@ -0,0 +1,43 @@
+name: Validate modified chains
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  golang-validate-modified:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Since we are using `pull_request_target` trigger (which runs in the context of the base of the pull request), we need to specify here to checkout the forked pr's code before running subsequent checks

Reply:
I found this article pretty useful https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/. I didn't yet convince myself that this pattern is going to be secure. 

 Can we prevent the CI jobs running until we have checked the PR over?